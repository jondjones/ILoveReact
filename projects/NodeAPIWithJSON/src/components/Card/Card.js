import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";

const MainImage = styled.div`
    img {
        object-fit: cover;
        width: 100%;
    }
`

const Title = styled.div`
    margin-top: 1rem;
    font-weight: bold;
`

const Item = styled.div`
    padding: 1rem;

    @media (min-width:961px) {
        margin-top: 5rem;
    }
`

const Content = styled.div`
    margin-top: 1rem;

    a {
        color: #6DE0D0;
        display: block;

        @media (min-width:961px) {
            padding-left: 1rem;
            display: inline-block;
        }
    }
` 

class Card extends Component {
  render() {
    
    const { title, description, image } = this.props;
    
    return (
        <Item>
            {image && 
                <MainImage>
                    <img src={image} alt={title} />
                </MainImage>
            }
            {title && 
                <Title>
                    {title}
                </Title>
            }
            {description && 
                <Content>
                    {description}
                    <a>Find Out More</a>
                </Content>

            }
        </Item>
    )}
}

Card.propTypes = {
    title: PropTypes.string,
    description: PropTypes.string,
    image: PropTypes.string
};

export default Card;