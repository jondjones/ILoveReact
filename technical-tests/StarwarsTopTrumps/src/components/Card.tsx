import React, { FunctionComponent } from 'react';
import styled from "styled-components"

const GridItem = styled.div`
    border: 1px solid white;
    padding-left: 20px;
    background-color: black;
    height: 30vh;
`
interface CardProps {}

const Card: FunctionComponent<CardProps> = (props) => {
  return (
    <GridItem>
        {props.children}
    </GridItem>
  )
}

export default Card;

