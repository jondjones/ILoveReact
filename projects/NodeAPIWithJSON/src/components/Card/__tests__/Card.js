import React from "react";
import { shallow, configure } from "enzyme";
import Card from "../Card";
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
describe("Item", () => {
  it("renders Item component", () => {
    const component = shallow(<Card />);
    expect(component).toMatchSnapshot();
  });
});
