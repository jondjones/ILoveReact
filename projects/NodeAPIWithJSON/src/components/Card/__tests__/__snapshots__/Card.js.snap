// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders Item component 1`] = `ShallowWrapper {}`;
